### DISCUSSION TITLE
Private reposetory in docker are a bad idea

### DISCUSSION THREAD

**Discussion by WeedInspector** [FieldRobotEvent/Virtual_Field_Robot_Event#54] [Show and tell]
Hi everyone,

we just want to leave the hint that its not a good idea to use private repository to build a docker image. 

Everything looks fine when you use `RUN git clone https://username:password@github.com/repository.git` in the dockerfile expect of the container dies immediately after start when using the `start_competition_environment.sh` magic. With `docker run` the container starts and the cloned repository is present and even the ros nodes are executable. 

Best regards and have fun
WeedInspector

**Comment by ThijsRuigrok** [comment]
I do not think the problem is related to the usage of a private repository. I think something is wrong with the start CMD used in your docker container. What start CMD did you use in the docker container? 

You can add a sleep command to your start command to make sure that your container stays alive, even when the start command fails to execute. The last line of your Dockerfile should then look like this:
`# start  your robot, and in the background sleep for 999999 seconds`
`CMD sh -c "roslaunch my_robot my_robot.launch --wait --screen; sleep 999999"`

After starting the competition environment, you can inspect what happened during the start by using running `docker logs fre_b_container_1`.  

Please let me know if this helped you.

**Reply by WeedInspector** [reply]
Sorry we somehow forgot the hint in our first post: After we only switched from a private to a public repo in the dockerfile to give it to you for debugging reasons everything worked as expected. 

So at the moment everything works fine. We just wanted to leave a hint about that.
